scipy/stats/_fit.py

```diff
@@ -782,12 +782,16 @@ def goodness_of_fit(dist, data, *, known_params=None, fit_params=None,
         to the Monte Carlo samples drawn from the null-hypothesized
         distribution. The purpose of these `guessed_params` is to be used as
         initial values for the numerical fitting procedure.
-    statistic : {"ad", "ks", "cvm", "filliben"}, optional
+    statistic : {"ad", "ks", "cvm", "filliben"} or callable, optional
         The statistic used to compare data to a distribution after fitting
         unknown parameters of the distribution family to the data. The
         Anderson-Darling ("ad") [1]_, Kolmogorov-Smirnov ("ks") [1]_,
         Cramer-von Mises ("cvm") [1]_, and Filliben ("filliben") [7]_
-        statistics are available.
+        statistics are available.  Alternatively, a callable with signature
+        ``(dist, data) ``may be supplied to compute the statistics.  Here
```

Comment:
```suggestion
        ``(dist, data)`` may be supplied to compute the statistics. Here
```

Reply:
All the other resampling functions that accept a vectorized statistic require it to accept an `axis` argument.

> `statistic` must also accept a keyword argument `axis` and be vectorized to compute the statistic along the provided axis of `data`.

They always pass it `-1`, but it gives us some flexibility and makes the code more explicit. Regardless of advantages, let's keep it consistent.